src/main/java/org/example/Phones.java

```diff
@@ -0,0 +1,107 @@
+package org.example;
+
+import java.util.*;
+
+public abstract class Phones {
+
+    List<Contact> contactList = new ArrayList<>();
+    HashMap<Contact, List<Message>> listMessageToContact = new HashMap<>();
+
+    List<String> callList = new ArrayList<>();
+    final String brand;
+    final String color;
+    final String material;
+
+    public double baterryLife;
+
+    public Phones(String brand, String color, String material, double bateryLife) {
+        this.brand = brand;
+        this.color = color;
+        this.material = material;
+        this.baterryLife = bateryLife;
+    }
+
+    public void createContact(Contact contact) {
+        if (contactList.contains(contact)) {
+            System.out.println("Contact exist already");
+        }
+        contactList.add(contact);
+    }
+
+
+
+    public void seeAllContacts() {
+        for (Contact contact : contactList) {
+            System.out.println("Name: " + contact.name + "/ PhoneNumber:" + contact.phoneNumber);
+        }
+    }
+
+
+
+    public void sendTextMessage(Contact contact, Message message) {
+        if (baterryLife < 0) {
+            System.out.println("The phone have no baterry");
+        } else {
+
+            if (contactList.contains(contact)) {
+                List<Message> messageList = listMessageToContact.get(contact);
+
+                if (messageList == null) {
+                    messageList = new ArrayList<>();
+                    listMessageToContact.put(contact, messageList);
+
+                }
+                messageList.add(message);
+                baterryLife = baterryLife - 2;
+            } else {
```

Comment:
you can use this as the first (guard) condition to simplify the condition. you have 3 levels of nested if/if/if.
refactor so that you don't perform any message sending if the battery is low or the contact doesn't exist